lib/router/utils.js

```diff
@@ -25,19 +25,9 @@ export function merge(hash, other) {
   Extracts query params from the end of an array
 **/
 export function extractQueryParams(array) {
-  let len = array && array.length,
-    head,
-    queryParams;
-
-  if (
-    len &&
-    len > 0 &&
-    array[len - 1] &&
-    array[len - 1].hasOwnProperty('queryParams')
-  ) {
-    queryParams = array[len - 1].queryParams;
-    head = slice.call(array, 0, len - 1);
-    return [head, queryParams];
+  let possibleQuery = array[array.length - 1];
+  if (possibleQuery && possibleQuery.hasOwnProperty('queryParams')) {
```

Comment:
I'd prefer to change this to `'queryParams' in possibleQuery` (or use `Object.prototype.hasOwnProperty.call(possibleQuery, 'queryParams')`).